docs/additional-examples/evm-swap.md

```diff
@@ -0,0 +1,656 @@
+      });
+    };
+```
+
+As everything on the blockchain works on hexadecimal hashes, we need to hash our transactions as well.
+
+```js
+    const hashTransaction = (tx) => {
+      return ethers.utils.arrayify(
+        ethers.utils.keccak256(
+          ethers.utils.arrayify(ethers.utils.serializeTransaction(tx)),
+        ),
+      );
+    }
+
+    Lit.Actions.setResponse({ response: "Conditions for swap not met!" });
+  }
+go();
+```
+
+A must `Lit.Actions.setResponse` needs to be set before we are closing the execution of a Lit Action. This is the response which is shown after the execution of a Lit Action.
+
+Complete Lit Action can be found at [swapAction.js](https://github.com/anshss/lit-evm-swap/blob/main/lit/swapAction.js).
+
+## 3) Implementing Mint/Grant/Burn with SDK
+
+### Installing Packages
+
+```bash
+npm i @lit-protocol/lit-node-client @lit-protocol/contracts-sdk 
+@simplewebauthn/browser  @lit-protocol/types ethers
+```
+
+### Imports Declarations
+
+```js
+import { LitNodeClient } from "@lit-protocol/lit-node-client";
+import { LitContracts } from "@lit-protocol/contracts-sdk";
+import { LitNetwork, AuthMethodScope, LIT_CHAINS } from "@lit-protocol/constants";
+import { LitAbility } from "@lit-protocol/types";
+import {
+    LitActionResource,
+    createSiweMessageWithRecaps,
+    generateAuthSig,
+    LitPKPResource,
+} from "@lit-protocol/auth-helpers";
+import { ethers } from "ethers";
+import bs58 from "bs58";
+```
+
+### Create Wallet instances
+
+These wallets will represent the two parties involved in the swap.
+
+```js
+async function getWalletA() {
+    const provider = new ethers.providers.JsonRpcProvider(
+        `https://yellowstone-rpc.litprotocol.com/`
+    );
+    const wallet = new ethers.Wallet(
+        process.env.NEXT_PUBLIC_PRIVATE_KEY_1,
+        provider
+    );
+    return wallet;
+}
+
+async function getWalletB() {
+    const provider = new ethers.providers.JsonRpcProvider(
+        `https://yellowstone-rpc.litprotocol.com/`
+    );
+    const wallet = new ethers.Wallet(
+        process.env.NEXT_PUBLIC_PRIVATE_KEY_2,
+        provider
+    );
+    return wallet;
+}c
+```
+
+### Create Lit Action and upload it to IPFS
+
+Once we are done with our Lit Action, we need to upload it to IPFS so we get an immutable CID which will always point to our action.
+
+```js
+export async function createLitAction() {
+    console.log("creating lit action..");
+    const action = createERC20SwapLitAction(chainAParams, chainBParams);
+    const ipfsCid = await uploadViaPinata(action);
+
+    console.log("Lit Action code:\n", action);
+    console.log("IPFS CID: ", ipfsCid);
+    return ipfsCid;
+}
+```
+
+### Create a Mint/Grant/Burn PKP
+We'll create a single function to perform 3 calls at once
+
+- Mint a PKP with the user's wallet
+- Add the Lit Action as a permitted auth Method
+- Burn the PKP NFT by transferring it to itself so auth methods become immutable
```

Comment:
> Burn the PKP NFT by transferring it to itself so auth methods become immutable

I think the important thing to emphasize here is that this method ensures that the PKP can only be used to sign based on the initial conditions that you define and nothing else

Reply:
which you do a good job of in the sentence below

Reply:
I can replace it to make it the same as the line 28
"Burn the PKP so that it can't be used to sign anything else other than what is outlined in the Lit Action logic"

The owner of pkp nft can send txn to smart contracts and modify auth methods so we burn it.
The primary reason for burning a pkp nft is to prevent any further modifications to the auth methods, even though the ultimate goal aligns with the above statement. I wrote this way to give more clarity to the developer on the thought process.

If it feels like not to be needed, I can replace it.

Reply:
this sounds good to me